File: app/src/main/java/com/wq/main/HomeFragment.java
package com.wq.main;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.sunrun.sunrunframwork.bean.BaseBean;
import com.wq.base.LBaseFragment;
import com.wq.common.model.LoginInfo;
import com.wq.common.quest.BaseQuestStart;
import com.wq.common.util.IntentUtil;
import com.wq.project01.R;
import com.wq.template.AddTemplateActivity;

import butterknife.BindView;
import butterknife.ButterKnife;

import static com.wq.common.quest.BaseQuestConfig.QUEST_LOGIN_CODE;

/**
 * 首页
 * Created by WQ on 2017/6/1.
 */

public class HomeFragment extends LBaseFragment {

    @BindView(R.id.tv_content)
    TextView tvContent;

    @Override
    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        //页面创建
        view.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                IntentUtil.startActivity(that, AddTemplateActivity.class);
            }
        });
        BaseQuestStart.login(this, "15916035572", "25d55ad283aa400af464c76d713c07ad");
    }

    public void nofityUpdate(int requestCode, BaseBean bean) {
        switch (requestCode) {
            case QUEST_LOGIN_CODE:
                //log 设置 tag为NetServer 可以查看请求情况
                tvContent.setText(bean.toString());
                if (bean.status == 200) {
                    LoginInfo info=bean.Data();//获取数据内容
                }
                break;
        }
        super.nofityUpdate(requestCode, bean);
    }

    @Override
    public int getLayoutRes() {
        return R.layout.fragment_home;
    }

}

File: app/src/main/java/com/wq/main/HomeFragment.kt
package com.wq.main

import android.os.Bundle
import android.view.View
import com.sunrun.sunrunframwork.adapter.ViewHodler
import com.sunrun.sunrunframwork.adapter.ViewHolderAdapter
import com.sunrun.sunrunframwork.bean.BaseBean
import com.sunrun.sunrunframwork.uiutils.PictureShow
import com.sunrun.sunrunframwork.utils.log.Logger
import com.wq.base.LBaseFragment
import com.wq.common.boxing.GlideMediaLoader
import com.wq.common.model.ClassTemplateBean
import com.wq.common.model.HomeTemplateBean
import com.wq.common.model.LoginInfo
import com.wq.common.quest.BaseQuestConfig.*
import com.wq.common.quest.BaseQuestStart
import com.wq.common.quest.Config
import com.wq.common.util.IntentUtil
import com.wq.main.adapter.HotTemplateAdapter
import com.wq.main.adapter.SearchHistoryAdapter
import com.wq.project01.R
import kotlinx.android.synthetic.main.fragment_home.*
import org.jetbrains.anko.support.v4.toast

/**
 * 首页
 * Created by WQ on 2017/6/1.
 */

class HomeFragment : LBaseFragment() {

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        //请求页面数据

        BaseQuestStart.getClassTemplate(this)
        BaseQuestStart.getShopRecommendTemplateList(this, "0", null, null)
        BaseQuestStart.getShopHotTemplateList(this, "0", null, null)
        tvSearch.setOnClickListener {
            IntentUtil.startTemplateListActivity(that, null, null, null)
//            toast("搜索")
        }
        hot_more.setOnClickListener {
            IntentUtil.startTemplateListActivity(that, GET_SHOP_HOT_TEMPLATE_LIST_URL, null, "热门模板")
//            toast("更多热门模板")
        }
        recommend_mre.setOnClickListener {
            IntentUtil.startTemplateListActivity(that, GET_GETSHOPRECOMMENDTEMPLATELIST_URL, null, "推荐模板")
//            toast("更多推荐模板")
        }
    }

    override fun nofityUpdate(requestCode: Int, bean: BaseBean) {
        when (requestCode) {
        //推荐模板
            QUEST_GETSHOPRECOMMENDTEMPLATELIST_CODE -> {
                Logger.D("${bean.data}")
                val data = bean.Data<List<HomeTemplateBean>>();
                recommendGridView.adapter = HotTemplateAdapter(that, data);
                recommendGridView.setOnItemClickListener { adapterView, view, index, l ->
                    IntentUtil.startTemplateDataActivity(that, data[index].template_name, data[index].template_id)
                }
            }
        //顶部分类.搜索历史
            QUEST_GET_CLASS_TEMPLATE_CODE -> {
                val info = bean.Data<List<ClassTemplateBean>>()//获取数据内容
                typeGrid.adapter = SearchHistoryAdapter(that, info, R.layout.item_tag_text);
                typeGrid.setOnItemClickListener { adapterView, view, i, l ->
                    IntentUtil.startTemplateListActivity(that, null,info[i].class_name,null)
                }
            }
        //热门模板
            QUEST_SHOP_HOT_TEMPLATE_LIST_CODE -> {
                val data = bean.Data<List<HomeTemplateBean>>();
                hotGridView.adapter = HotTemplateAdapter(that, data)
                hotGridView.setOnItemClickListener { adapterView, view, index, l ->
                    IntentUtil.startTemplateDataActivity(that, data[index].template_name, data[index].template_id)
                }
            }

        }
        super.nofityUpdate(requestCode, bean)
    }

    override fun getLayoutRes(): Int {
        return R.layout.fragment_home
    }

}
